distributed/client.py

```diff
@@ -4053,8 +4053,64 @@ class will be instantiated with any extra keyword arguments.
         if isinstance(plugin, type):
             plugin = plugin(**kwargs)
 
+        if name is None:
+            name = _get_worker_plugin_name(plugin)
+
+        assert name
+
         return self.sync(self._register_worker_plugin, plugin=plugin, name=name)
 
+    async def _unregister_worker_plugin(self, name):
+        responses = await self.scheduler.unregister_worker_plugin(name=name)
+
+        for response in responses.values():
+            if response["status"] == "error":
+                exc = response["exception"]
+                tb = response["traceback"]
+                raise exc.with_traceback(tb)
+        return responses
+
+    def unregister_worker_plugin(self, name):
+        """Unregisters a lifecycle worker plugin
+
+        This unregister a worker plugin using its name attribute by calling
+        the plugin's teardown method. (See the ``dask.distributed.WorkerPlugin``
+        class or the examples below for the interface and docstrings.)
+
+        Parameters
+        ----------
+        name: str
```

Comment:
Nitpick: `distributed` (and `dask`) use [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) for formatting docstrings. numpydoc's style guide has a space between the parameter name and colon here. 

```suggestion
        name : str
```